modules/glslang/register_types.cpp

```diff
@@ -161,6 +161,11 @@ static Vector<uint8_t> _compile_shader_glsl(RenderingDevice::ShaderStage p_stage
 	spv::SpvBuildLogger logger;
 	glslang::SpvOptions spvOptions;
 
+	if (Engine::get_singleton()->is_optimize_spirv_enabled()) {
+		spvOptions.disableOptimizer = false;
```

Comment:
I would add a GLOBAL_DEF to turn it on and off, this way testing is more easy and it can be overridden by platform (as in, probably will be needed on Android).

Reply:
That sounds fine to me, although we'd likely need to force a restart for it to take effect properly (as it won't look for shaders again). I guess it should be GLOBAL_DEF_RST instead.